### PR TITLE
8275868: ciReplay: Inlining fails with "unloaded signature classes" due to wrong protection domains

### PR DESCRIPTION
**Pull request by GoeLin**
I had to do some trivial resolves. 

I add some refactorings from "8262912: ciReplay: replay does not simulate unresolved classes"
to the edited test files to make the test compile. I also had to fix the path to WhiteBox.

I also added two methods to query names to ciEnv from
"8271911: replay compilations of methods which use JSR292 (easy cases)".
this calls further into dyno_name(). I disabled this, it is not needed as long as 8262912 is not backported.

Test reproduces the issue as expected.

For follow up 8276227 see dependent pull request.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] [JDK-8275868](https://bugs.openjdk.org/browse/JDK-8275868) needs maintainer approval
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8275868](https://bugs.openjdk.org/browse/JDK-8275868): ciReplay: Inlining fails with "unloaded signature classes" due to wrong protection domains (**Bug** - P4 - Approved)

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev.git pull/2334/head:pull/2334` \
`$ git checkout pull/2334`

Update a local copy of the PR: \
`$ git checkout pull/2334` \
`$ git pull https://git.openjdk.org/jdk17u-dev.git pull/2334/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2334`

View PR using the GUI difftool: \
`$ git pr show -t 2334`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/2334.diff">https://git.openjdk.org/jdk17u-dev/pull/2334.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk17u-dev/pull/2334#issuecomment-2022264682)

### CODE REVIEW

File: src/hotspot/share/ci/ciEnv.cpp [hunk 1]
```diff
@@ -1185,6 +1185,26 @@ ciInstance* ciEnv::unloaded_ciinstance() {
...
+  if (k->is_instance_klass()) {
+    return replay_name(k->as_instance_klass()->get_instanceKlass());
+  }
+  return k->name()->as_quoted_ascii();
+}
+
+// Look up the location descriptor for the given class and return it as a string.
+// Returns the class name as a fallback if no location is found.
+const char *ciEnv::replay_name(const InstanceKlass* ik) const {
+  const char* name = nullptr; // dyno_name(ik);
```
Comment (TheRealMDoerr) [inline src/hotspot/share/ci/ciEnv.cpp:1201]: I'd appreciate to have a comment why `dyno_name(ik)` is commented out. Maybe "JDK-8271911 is not in JDK 17, so we fall back to using the class name below."

**Review by TheRealMDoerr — COMMENTED** [review]
Looks basically good. Please see my suggestions.

File: test/hotspot/jtreg/compiler/ciReplay/CiReplayBase.java [hunk 2]
```diff
@@ -156,13 +160,13 @@ public boolean generateReplay(boolean needCoreDump, String... vmopts) {
             options.add(needCoreDump ? ENABLE_COREDUMP_ON_CRASH : DISABLE_COREDUMP_ON_CRASH);
             if (needCoreDump) {
                 // CiReplayBase$TestMain needs to be quoted because of shell eval
-                options.add("-XX:CompileOnly='" + TestMain.class.getName() + "::test'");
+                options.add("-XX:CompileOnly='" + TestMain.class.getName() + "::" + getTestMethod() + "'");
```
[the remark this answers is not in the document]
Comment (GoeLin) [reply test/hotspot/jtreg/compiler/ciReplay/CiReplayBase.java:163]: I wanted to leave that to the potential backport of 8262912.  It's not needed for the purpose of this change. Same below.

**Review by TheRealMDoerr — APPROVED** [review]
Ok. Thanks for the update.

**Pull request closed without merging**